storagemarket/deal_handler_test.go

```diff
@@ -0,0 +1,64 @@
+package storagemarket
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCancellationSimple(t *testing.T) {
```

Comment:
@aarshkshah1992 could we add some godoc around these tests to explain what they are testing.

There is also godoc missing in the `dealHandler` so it is not clear who/why calls `dh.cancel` and `dh.transferCancelled`.

Reply:
done.